### PR TITLE
refactor: add docsService for blocksuite to change and observe doc mode

### PR DESCRIPTION
**Pull request by akumatus**
This PR provide `DocModeService` to manage document mode, which is used by blocksuite.

### What Changed?
- Add document mode apis from `DocService` and `DocsService`
- Remove useless `pageService.getEditorMode` and `pageService.slots.editorModeSwitch` api

[Related PR in blocksuite](https://github.com/toeverything/blocksuite/pull/7288)

[Bump Blocksuite](https://github.com/toeverything/AFFiNE/pull/7209)
> ## Features
> * [feat(presets): support different tracker control for actions blocksuite#7285](https://github.com/toeverything/blocksuite/pull/7285) @donteatfriedrice
> * [feat(edgeless): add default color and editing border for edgeless text blocksuite#7278](https://github.com/toeverything/blocksuite/pull/7278) @Flrande
> * [feat(blocks): center peek entries blocksuite#7275](https://github.com/toeverything/blocksuite/pull/7275) @pengx17
> 
> ## Bugfix
> * [fix(blocks): affine-embed-synced-doc-block peek on dblclick blocksuite#7284](https://github.com/toeverything/blocksuite/pull/7284) @pengx17
> * [fix(edgeless): edgeless text rect should automatically update when changing font blocksuite#7292](https://github.com/toeverything/blocksuite/pull/7292) @Flrande
> * [fix(edgeless): align panel not work blocksuite#7293](https://github.com/toeverything/blocksuite/pull/7293) @Flrande
> * [fix(blocks): filterable list unexpected hover style blocksuite#7291](https://github.com/toeverything/blocksuite/pull/7291) @golok727
> * [fix(blocks): slash menu unexpected hover style blocksuite#7290](https://github.com/toeverything/blocksuite/pull/7290) @golok727
> * [fix(blocks): bookmark block error on undo or delete while loading blocksuite#7276](https://github.com/toeverything/blocksuite/pull/7276) @golok727
> * [fix(presets): chat input max-height blocksuite#7283](https://github.com/toeverything/blocksuite/pull/7283) @fundon
> * [fix(presets): get image blob from selection blocksuite#7286](https://github.com/toeverything/blocksuite/pull/7286) @donteatfriedrice
> * [fix(blocks): connectors resize and rotate blocksuite#7256](https://github.com/toeverything/blocksuite/pull/7256) @fundon
> 
> ## Refactor
> * [refactor: add DocModeService for doc mode get, set, toogle and observe change blocksuite#7288](https://github.com/toeverything/blocksuite/pull/7288) @akumatus
> * [refactor(edgeless): rewrite auto-connect component as a widget blocksuite#7287](https://github.com/toeverything/blocksuite/pull/7287) @doouding
> 
> ## Misc
> * [test: fix flaky blocksuite#7295](https://github.com/toeverything/blocksuite/pull/7295) @L-Sun
> * [style(page): adjust heading font-weight and letter-spacing blocksuite#7277](https://github.com/toeverything/blocksuite/pull/7277) @CatsJuice
> * [chore: apply member ordering rule blocksuite#7279](https://github.com/toeverything/blocksuite/pull/7279) @Saul-Mirone

---

### CODE REVIEW

**Comment by akumatus** [comment]
* **#7200** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7200?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — APPROVED** [review]
LGTM

File: packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx [hunk 1]
```diff
@@ -21,12 +21,16 @@ const EditorChatPanel = ({ editor }: SidebarTabProps) => {
...
-      (chatPanelRef.current as ChatPanel).doc = editor.doc;
-    });
-    pageService.slots.editorModeSwitch.on(() => {
-      (chatPanelRef.current as ChatPanel).host = editor.host;
-    });
+    const disposable = [
+      pageService.slots.docLinkClicked.on(() => {
+        (chatPanelRef.current as ChatPanel).doc = editor.doc;
+      }),
+      pageService.docsService.onModeChange(() => {
```
Comment (L-Sun) [inline packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx]: `docsService` -> `docModeService`

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/spec-patchers.tsx [hunk 2]
```diff
@@ -281,6 +282,57 @@ export function patchPeekViewService(
...
+    spec => spec.schema.model.flavour === 'affine:page'
+  ) as BlockSpec<string, RootService>;
+
+  if (!rootSpec) {
+    return specs;
+  }
+
+  patchSpecService(rootSpec, pageService => {
+    const DEFAULT_MODE = 'page';
+    pageService.docModeService = {
```
Comment (L-Sun) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/spec-patchers.tsx:300]: This error may be resolved after bumping bs

File: packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx (hunk 1 above)
Comment (akumatus) [reply packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx]: Thanks for your comment! That is a mistake and I will fix it.

File: packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx [hunk 3]
```diff
@@ -21,12 +21,16 @@ const EditorChatPanel = ({ editor }: SidebarTabProps) => {
...
-    });
-    pageService.slots.editorModeSwitch.on(() => {
-      (chatPanelRef.current as ChatPanel).host = editor.host;
-    });
+    const disposable = [
+      pageService.slots.docLinkClicked.on(() => {
+        (chatPanelRef.current as ChatPanel).doc = editor.doc;
+      }),
+      pageService.docModeService.onModeChange(() => {
+        (chatPanelRef.current as ChatPanel).host = editor.host;
```
Comment (L-Sun) [inline packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx:30]: This `editor.host` may be `undefined` if you switch the editor mode. You can reproduce this issue with the following steps:
1. Open a document.
2. Open the chat panel on the right side.
3. Switch the editor mode.

**Review by Saul-Mirone — APPROVED** [review]

**Review by CatsJuice — APPROVED** [review]

File: packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx (hunk 3 above)
Comment (akumatus) [reply packages/frontend/core/src/modules/multi-tab-sidebar/multi-tabs/tabs/chat.tsx:30]: Thanks! This is a very good point I overlooked. 

The `editor.host` type is forced as `EditorHost` in blocksuite, which might be undefined. I think we should fix this in another PR.

image
 
I have fixed this problem by detect the existence of `editor.host` and everything works fine, cause the chat panel will re-render affter mode switched and the mode$ works like behavior subject.

**Pull request merged**